HSLdevcom/bultti-ui
# 333 role fixes according to role matrix

Closes https://github.com/HSLdevcom/bultti/issues/333

Note: currently only admins can accept pre and post inspections. We'll implement double accept later where also operator can accept their owns inspections

Questions:
* Should I remove those methdos marked with TODO: remove? Note: I don't want this PR to be merged with any new TODO's so something needs to be done with them (I found like 5-8 unused resolvers)
* Is guardPlannedUnitExecutionRequirementResolvers OK that it has allowedRoles: [UserRole.HSL]?
* refreshEquipmentInCatalogueMutation: this was OK to be available for all roles
* Shouldn't we remove DEV Load sanctions button from post inspection's page? We have devResolver for that, if ever want to test running some functions in backend.
* "Sopimuksessa mukana toggle" - is the endpoint of this now correctly restricted by role checks? I guess admins and operators can do it or am I right? Can't easilly find which resolver this checkbox calls....

How to test this PR:

* Test every action/resolver with every role (admin, hsl user, operator). Ensure that each action has access rights according to our access right table: https://gitlab.hsl.fi/bultti/bultti-testaus/bultti/-/wikis/Ka%CC%88ytto%CC%88oikeudet
* You should also test opening post inspections that are in DRAFT with operator and HSL user - shouldn't work.
* Also you shouldn't be able to open pre inspections in DRAFT with HSL user
